tfhe/src/c_api/high_level_api/integers.rs

```diff
@@ -659,3 +659,112 @@ pub unsafe extern "C" fn compact_fhe_int256_list_try_encrypt_with_compact_public
 }
 
 define_all_cast_into_for_integer_type!(FheBool);
+
+macro_rules! impl_oprf_for_uint {
+    (
+        name: $name:ident
+    ) => {
+
+        ::paste::paste! {
+            #[no_mangle]
+            pub unsafe extern "C" fn [<par_generate_oblivious_pseudo_random_ $name:snake>](
+                out_result: *mut *mut $name,
+                seed_low_bytes: u64,
+                seed_high_bytes: u64,
+                random_bits_count: u64,
+            ) -> c_int {
+
+                $crate::c_api::utils::catch_panic(|| {
+                    let seed_low_bytes: u128 = seed_low_bytes.into();
+                    let seed_high_bytes: u128 = seed_high_bytes.into();
+                    let seed = concrete_csprng::seeders::Seed((seed_high_bytes << 64) | seed_low_bytes);
+
+                    let result = crate::high_level_api::par_generate_oblivious_pseudo_random_unsigned_integer(seed, random_bits_count);
+                    *out_result = Box::into_raw(Box::new($name(result)));
+                })
+            }
+        }
+    };
+}
+
+macro_rules! impl_oprf_for_int {
+    (
+        name: $name:ident
+    ) => {
+
+        ::paste::paste! {
+            #[no_mangle]
+            pub unsafe extern "C" fn [<par_generate_oblivious_pseudo_random_unsigned_ $name:snake>](
+                out_result: *mut *mut $name,
+                seed_low_bytes: u64,
+                seed_high_bytes: u64,
+                random_bits_count: u64,
+            ) -> c_int {
+                $crate::c_api::utils::catch_panic(|| {
+                    let seed_low_bytes: u128 = seed_low_bytes.into();
+                    let seed_high_bytes: u128 = seed_high_bytes.into();
+                    let seed = concrete_csprng::seeders::Seed((seed_high_bytes << 64) | seed_low_bytes);
+
+                    let result =
+                        crate::high_level_api::par_generate_oblivious_pseudo_random_signed_integer(
+                            seed,
+                            crate::high_level_api::SignedRandomizationSpec::Unsigned {
+                                random_bits_count
+                            },
+                        );
+                    *out_result = Box::into_raw(Box::new($name(result)));
+                })
+            }
+        }
+
+        ::paste::paste! {
+            #[no_mangle]
+            pub unsafe extern "C" fn [<par_generate_oblivious_pseudo_random_full_signed_ $name:snake>](
```

Comment:
full -> full_range maybe ? Really unsure, do we know if cbindgen is able to grab the doc from rust ? could be a nice thing to explain what functions do

Reply:
Sadly, docs written in macro are not picked up by cbindgen, even when they are in the form of attribute `#[doc("This function does this")]`, having doc in the C header is something that I'd like to be able to have, but haven't managed to

Reply:
well now we have a fork of cbindgen, I have archived it for now (unless cbindgen PRs get some attention, but it feels unlikely) so if C docs in cbindgen haunt your weekends we could add it to our fork 😅